### PR TITLE
enhance: Create/Delete support of namespace

### LINKED ISSUE #20: Enhance: Create/Delete support of namespace
Most users will specify the namespace as part of any object they create or delete, therefore the namespace parameter should be removed or the namespace should be extracted from the object and take precedence over the value in the object.

### PR DESCRIPTION
**Pull request by shad7**
Adds use of namespace that is included inside the object to
create or delete.
Includes namespace validation based on the current rules in use
within Kubernetes.

closes #20

### CODE REVIEW

File: kubeshift/validator.py [hunk 1]
```diff
@@ -1,41 +1,61 @@
...
+    if not re.match('(?:http|https)://', url):
+        raise KubeShiftError('URL does not include HTTP or HTTPS')
+
+
+def is_valid_ns(ns):
+    """Verify namespace format."""
+    valid = True
+    if not ns:
+        valid = False
+    elif len(ns) > 63:
```
Comment (cdrage) [inline kubeshift/validator.py]: Bit confused what this does here, 63 a max length for namespaces?

Comment (cdrage) [inline kubeshift/validator.py]: Comment here explaining why may be best :)

Comment (shad7) [inline kubeshift/validator.py]: That is what is within the validation done by kubernetes.
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/validation/validation.go

Comment (cdrage) [inline kubeshift/validator.py]: Ahhhh makes sense!

**Pull request merged**